web-common/src/features/charts/render/vega-signals.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Util methods for handling vega signals
+ */
+
+export function resolveSignalField(value: unknown, field: string) {
+  if (typeof value === "object" && value !== null) {
+    return value[field]?.[0];
+  }
+  return undefined;
```

Comment:
**Security Issue**: The function 'resolveSignalField' does not validate the type of the 'field' parameter, which could lead to unintended property access if a non-string value is passed. <br> **Fix**: Add a type check for the 'field' parameter to ensure it is a string. <br> **Code Suggestion**: 
 ```
 +  if (typeof field !== 'string') throw new TypeError('Field must be a string');
 +  if (typeof value === "object" && value !== null) {
 +    return value[field]?.[0];
 +  }
 +  return undefined;
 ```